### PR TITLE
RFC: Configurable access groups

### PR DESCRIPTION
**Pull request by bpedersen2**
Introduce easier overridabel compile time configuration and use it for a configurable accessgroups service.

## Motivation

Configuring the access groups service needed a custom service provider factory. This needed to get copied during the build, making it hard to validate for all facilities. 

This change introduces a new override config file, that can be used  for injecting advanced configuration objects that are not environment vars( but as this is merged, allows to keep most defaults easily).

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ needs manual test for graphql ] Passing? (Merge will not be approved unless this is checked) 
- [ no ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

### CODE REVIEW

File: src/config/configuration.ts [hunk 1]
```diff
@@ -25,6 +28,17 @@ const configuration = () => {
...
     },
+    accessGroupsStaticConfig: {
+      enabled: boolean(process.env?.ACCESS_GROUPS_STATIC_ENABLED || true),
+      value: accessGroupsStaticValues.split(",").map((v) => v.trim()) ?? [],
+    },
+    accessGroupsOIDCPayloadConfig: {
+      enabled: boolean(process.env?.ACCESS_GROUPS_OIDCPAYLOAD_ENABLED || false),
+    },
+    accessGroupsGraphQlConfig: {
+      enabled: boolean(process.env?.ACCESS_GROUPS_OIDCPAYLOAD_ENABLED || false),
```
Comment (bpedersen2) [inline src/config/configuration.ts]: fix: use  GRAPHQL!

**Comment by bpedersen2** [comment]
Needs local testing at ESS ( as I don't have a graphql endpoint available). 

Set 
```
ACCESS_GROUPS_GRAPHQL_ENABLED=true
ACCESS_GROUP_SERVICE_HANDLER=../../config/graphqlHandler
```

File: src/config/localconfiguration.ts
Comment (nitrosx) [inline src/config/localconfiguration.ts:1]: Can you explain why you would want a local configuration file?
I'm not oppose, just curious.

**Review by nitrosx — COMMENTED** [review]
Please see my comment on localconfiguration

File: src/config/localconfiguration.ts
Comment (nitrosx) [reply src/config/localconfiguration.ts:1]: We should probably include in documentation the structure of the configuration file and provide a template

**Comment by nitrosx** [comment]
@bpedersen2 can you let me know how I can test this feature before I merge it? Thanks

**Comment by nitrosx** [comment]
It has been merged in PR #1041.
Most likely it has become obsolete and can be closed.
I will check with @Junjiequan and @bpedersen2  first

**Comment by bpedersen2** [comment]
Closing as it is now part of #1041

**Pull request closed without merging**